Batch Newsletter Service

laurenp-2 opened the pull request:
### Summary <!-- Required -->

<!-- Provide a general summary of your changes in the Title above -->
<!-- Itemize bug fixes, new features, and other changes -->
<!-- Feel free to break this into sub-sections, i.e. features, fixes, etc. -->
<!-- Some examples are shown below. -->

This pull request is the first step towards implementing the email service for batch newsletter distribution. 

#### Completed features: 
- [x] Implemented a emailService that: retrieves users from Firebase in batches of 50, sends out emails to each batch using the Resend API. This function also supports single send emails (to a specified email address) to allow for testing of an email campaign before it is released. 
- [x] Implemented a newsletter email template that allows easy customization. 
- [x] Implemented a desktop and mobile view version of the newsletter (only supported in certain clients, such as Apple Mail). While the newsletter template is written to be as globally responsive as possible, note that appearance varies among different email services; a newsletter may appear differently in Gmail vs Yahoo Mail vs Outlook. 


### Test Plan <!-- Required -->
The backend route was tested using my email (so that no email campaigns were sent to CUApts users). 

This is how a received Newsletter email appears on desktop (Gmail): 
<img width="350" height="700" alt="desktop 1" src="https://github.com/user-attachments/assets/0cfb87ac-d198-47f2-8f3a-804996635e98" />
<img width="314" height="699" alt="desktop 2" src="https://github.com/user-attachments/assets/55cf9486-f771-4bbc-a298-0760cd282c33" />
<img width="385" height="698" alt="desktop 3" src="https://github.com/user-attachments/assets/83df106d-5688-4803-9bab-fec7b127b18c" />


This is how a received newsletter appears on mobile (Apple mail app): 
<img width="350" alt="mobile 1" src="https://github.com/user-attachments/assets/89ff8bc6-aa4a-4343-bdd7-0123da651960" />
<img width="350" alt="mobile 2" src="https://github.com/user-attachments/assets/4057a246-2259-480e-974f-6006c38dc8e8" />

**Note:** Email clients render HTML and CSS differently, so emails may appear slightly different across inboxes. For example, Apple Mail supports custom fonts, while Gmail does not. The images above may also look different depending on the email client. Please test any emails that are being sent out vigorously among different clients to ensure consistent and responsive results!

Commit:
Fix ESLint and TypeScript configuration

- Update backend/.eslintrc.json to reference the correct tsconfig.json path
- Enable JSX support in tsconfig.json with "jsx": "react-jsx"

Commit:
Add Resend and React-Email as dependencies

Commit:
Implement basic email service

- Create email sending scripts
- Create basic components for newsletter

Commit:
Merge branch 'main' of https://github.com/cornell-dti/cu-apts into newsletter

Commit:
Add Email Icons

- Create assets folder in email directory
- Add email icons to assets

Commit:
Implement basic Newsletter

- Create Newsletter templates
- Create components for newsletter sections
- Create generateNewsletter component to allow customization
- Update README (WIP)
- Update emailService to mail newsletter created by GenerateNewsletter

Commit:
Adjust TS Parsing

Commit:
Add React-Icons

Commit:
Fix parsing errors

- Edit tsconfig to allow JSX in react files
- Add working directory configuration in eslintrc to prevent parsing errors

- TODO: figure out why these errors keep on occuring

Commit:
Implement Newsletter template

- Create ReelsFeature, SubleaseSpotlight, FeatureSpotlight and NeighborhoodComparison components
- Edit styling of existing newsletter components
- Add prop types for new components to Types.tsx
- Add new components to Newsletter template
- Add customization options for new components in GenerateNewsletter

Commit:
Implement Basic Newsletter (Web)

- Implement all components of Newsletter
- Render property cards in AreaSpotlight and LandlordHighlight (based on property id)

- TODO: finish styling, add documentation

Commit:
Add documentation

- Add top-level documentation to all Newsletter components

Commit:
Implement batch sending

- Load firebase users in batches of 50
- Implement script to send emails out to all users in loaded batches

Commit:
Merge branch 'main' of https://github.com/cornell-dti/cu-apts into newsletter

Commit:
Add documentation in GenerateNewsletter

Commit:
Provide documentation for email system

- Update README for email directory
- Update import in firebase_users_loader

Commit:
Update email system README

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cornell-dti/cu-apts?pullRequest=404) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cornell-dti/cu-apts?pullRequest=404) before we can accept your contribution.<br/><hr/>**Lauren Pothuru** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cornell-dti/cu-apts?pullRequest=404) it.</sub>

Commit:
Add option to send email to one address

Commit:
Debug property fetching

- Fix relative vs absolute path in property fetch
- Add info to README

Commit:
Style newsletter components

CasperL1218 reviewed (dismissed):
This PR is comprehensive with a lot of detailed carried out well. Here are some main points to note:

1. The README file is very informative and thorough. The "important" notices are helpful!
2.  For the styling of component, you could organize them with useStyle, which could make styling more intuitive and clear, and mainly more reusable. You could checkout the other files to see how it is done. (this is more of a code pattern issue, this could be done next semester on a separate task by you or someone else so no worries now)
3. For backend/scripts/email/emailService.ts file, you have some commented out logic from line 84 to 92. It would be nice to remove them if they are no longer needed

Commit:
Merge branch 'main' of https://github.com/cornell-dti/cu-apts into newsletter

Commit:
Update email service and README for newsletter campaign
- Abstract unnecessary comments and clean up code
- Fix environment variable handling in firebase config
- Update README for clarity and formatting

Commit:
Add mobile responsiveness and refactor email service

- Update image icons for better compatibility
- Add mobile header; prevent dark mode (only supported in some email clients)
- Abstract email sending into separate functions

Commit:
Remove LandlordHighlight component and related props from newsletter template

CasperL1218 commented on backend/scripts/email/emailService.ts:
```diff
@@ -0,0 +1,216 @@
+    return;
+  }
+
+  // const { API_BASE_URL } = process.env;
```
small issue, just remember to check and remove unused/commented-out lines

CasperL1218 commented on backend/scripts/email/emailService.ts:
Great implementation with good structured code and layout. 
After conducting the testings, you could organize the test cases in a separate testing files to better separate feature and test files.

CasperL1218 reviewed (changes requested):
Great implementation of the html template with the approach to resolve the preloading issues. 
Some small details to clean up wrapping up include a brief check of removing unused/commented-out sections, organizing testing functions into a separata testing file for code quality and organization. 

Looking forward to the new design updates and the admin end input! Good job!

Commit:
Implement May's Redesign
